In [1]:
d<-read.csv('Autoparts.csv')
attach(d)

In [2]:
m1 <- lm(Sales~Mktg, data=d);

In [5]:
summary(m1)


Call:
lm(formula = Sales ~ Mktg, data = d)

Residuals:
    Min      1Q  Median      3Q     Max 
-36.714 -14.685  -6.855  18.193  53.426 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)   
(Intercept) 77.383975  21.104050   3.667  0.00208 **
Mktg        -0.008265   0.059436  -0.139  0.89114   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 29.31 on 16 degrees of freedom
Multiple R-squared:  0.001207,	Adjusted R-squared:  -0.06122 
F-statistic: 0.01934 on 1 and 16 DF,  p-value: 0.8911


The Findings:

1. The relationship between the variables Sales and Mktg is negatively correlated and not significant.
2. The model only explains 0.12% of the data.
3. The p-value is 0.8911 which is very low, that indicates that the model is poor.

In [18]:
m2 <- lm(Sales~Mktg+Outlets+Popn+Vehicles+Reps, data=d);
summary(m2);


Call:
lm(formula = Sales ~ Mktg + Outlets + Popn + Vehicles + Reps, 
    data = d)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.1716  -1.4581   0.0432   2.1715   5.0267 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.582457   5.414913   1.954  0.07437 .  
Mktg         0.036144   0.009645   3.747  0.00278 ** 
Outlets      0.131756   0.015815   8.331 2.48e-06 ***
Popn        -6.050930   2.668976  -2.267  0.04266 *  
Vehicles     6.527655   4.480298   1.457  0.17079    
Reps        -0.041927   0.766622  -0.055  0.95729    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 4.179 on 12 degrees of freedom
Multiple R-squared:  0.9848,	Adjusted R-squared:  0.9784 
F-statistic: 155.3 on 5 and 12 DF,  p-value: 1.789e-10


The Findings:

1. Outlets become the most significant variable for sales, but the impact is not so high.
2. Population has less significant but negative impact on sales
3. This model is relatively better in terms of R-squared. It explains more than 98% of the data.

In [26]:
rsq_m1<-summary(m1)$r.squared
rsq_m2<-summary(m2)$r.squared

adj_rsq_m1<-summary(m1)$adj.r.squared
adj_rsq_m2<-summary(m2)$adj.r.squared

aic_m1<-AIC(m1)
aic_m2<-AIC(m2)

bic_m1<-BIC(m1)
bic_m2<-BIC(m2)

In [27]:
mat_table <- matrix(c(rsq_m1, adj_rsq_m1, aic_m1, bic_m1, rsq_m2, adj_rsq_m2, aic_m2, bic_m2),ncol=4,byrow=TRUE)
colnames(mat_table) <- c("R-Squared","Adjusted R-Squared","AIC", "BIC")
rownames(mat_table) <- c("Model 1","Model 2")

In [28]:
mat_table

R-Squared   Adjusted R-Squared AIC      BIC     
Model 1 0.001207045 -0.06121751        176.5729 179.2441
Model 2 0.984777594  0.97843493        109.2649 115.4975

The Value of AIC and BIC is relatively lower for Model 2 which indicates that the Model 2 is better than the Model 1.